### PR TITLE
Replace backslashes with Path.DirectorySeparatorChar in SettingsUtility

### PR DESCRIPTION
**Pull request by akoeplinger**
NuGet would create a folder `~/.nuget\packages\` on non-Windows machines before.

Fixes https://github.com/NuGet/Home/issues/1517

### CODE REVIEW

File: src/NuGet.Core/NuGet.Configuration/Utility/SettingsUtility.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ public static class SettingsUtility
         public const string ConfigSection = "config";
         public const string GlobalPackagesFolderKey = "globalPackagesFolder";
         public const string GlobalPackagesFolderEnvironmentKey = "NUGET_PACKAGES";
-        public const string DefaultGlobalPackagesFolderPath = ".nuget\\packages\\";
+        public const string DefaultGlobalPackagesFolderPath = ".nuget/packages/";
```
Comment (yishaigalatzer) [inline src/NuGet.Core/NuGet.Configuration/Utility/SettingsUtility.cs]: Can you explain why this change is necessary, rather than do the replace below in the opposite direction?

I would instead probably use a path.combine to generate a readonly string on the fly?

Comment (akoeplinger) [inline src/NuGet.Core/NuGet.Configuration/Utility/SettingsUtility.cs]: @yishaigalatzer I did it this way to be consistent with what you do on Line 146.

Personally, I like the readonly string approach better so I've changed it to that.

**Comment by yishaigalatzer** [comment]
sweet :shipit: 
@emgarten can you take a look and if it builds/passes tests merge it?

**Comment by emgarten** [comment]
nuget/nuget.client@4b24d812ae2164dfbb9b62dd44a1d61b6efb6e3b
